api/src/methods/new_payload.rs

```diff
@@ -384,15 +384,27 @@ mod tests {
         let head_hash = B256::new(hex!(
             "781f09c5b7629a7ca30668e440ea40557f01461ad6f105b371f61ff5824b2449"
         ));
-        let genesis_block = Block::default()
-            .into_extended_with_hash(head_hash)
-            .with_value(U256::ZERO);
+        let genesis_block = {
+            let mut tmp = Block::default();
+            tmp.header.state_root = genesis_config.initial_state_root;
```

Comment:
Is it done for the same reason as what the comment says in `app::command::forkchoice_state()`, i.e. to propagate full move state to the app?